eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/UpdateSortKeysCommand.php

```diff
@@ -61,7 +61,8 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         /** @var \eZ\Publish\API\Repository\Repository $repository */
         $repository = $this->getContainer()->get('ezpublish.api.repository');
-        $repository->setCurrentUser(
+        $permissionResolver = $this->getContainer()->get('eZ\Publish\API\Repository\PermissionResolver');
```

Comment:
```suggestion
        $permissionResolver = $this->getContainer()->get(PermissionResolver::class);
```

or just inject permission resolver into command.

Reply:
Not sure if this a correct way, as `eZ\Publish\API\Repository\PermissionResolver` refers to service name and not class per se.

Reply:
> Not sure if this a correct way, as `eZ\Publish\API\Repository\PermissionResolver` refers to service name and not class per se.

The whole point of switching to FQCN-named services is the ability to use `::class` notation.

Reply:
FWIW: I am generally not fond of FQCN-named services for platform/library code.
The whole point of using a service name instead of a class name is to abstract away the service implementation from its interface. And in any given eZ project, I do generally reimplement a few services from the Kernel, to add fixes or extra capabilities. Using the FQCN of the original class does not help me in that case, as it kinda hints that the service in use is still the original eZ one, whereas it is not any more...
I'd leave usage of QCN-named services to app code, unless there are other big advantages to that (I discount the fact that using class names helps autocomplete in IDEs as a big advantage, as autocomplete-based coding leads to frequent and sometimes subtle mistakes).

Reply:
> FWIW: I am generally not fond of FQCN-named services for platform/library code.
> The whole point of using a service name instead of a class name is to abstract away the service implementation from its interface. And in any given eZ project, I do generally reimplement a few services from the Kernel, to add fixes or extra capabilities. Using the FQCN of the original class does not help me in that case, as it kinda hints that the service in use is still the original eZ one, whereas it is not any more...

Please use your own implementation FQCN name along with [`decorates`](https://symfony.com/doc/3.4/service_container/service_decoration.html) attribute inside Symfony DIC.

Reply:
Sorry for the delay, I was off.

> `decorates` is useful, but that means that my own code will be using one service name which is different from the name used by the kernel to refer to the same service. 

No, it doesn't :-)
One of the advantages of doing e.g.:
``` yaml
services:
    App\MyResolver:
        decorates: eZ\Publish\API\Repository\PermissionResolver
        arguments:
            $innerPermissionResolver: '@App\MyResolver.inner'
```
``` php
class MyResolver implements PermissionResolver
{
    /** @var \eZ\Publish\API\Repository\PermissionResolver */
    private $innerPermissionResolver;

    public function __construct(PermissionResolver $innerPermissionResolver)
    {
        $this->innerPermissionResolver = $innerPermissionResolver;
    }
    // ...
}
```
is that when injecting, either explicitly or by auto-wiring, the original service:

``` php
class MyService
{
    /** @var \eZ\Publish\API\Repository\PermissionResolver */
    private $permissionResolver;

    public function __construct(PermissionResolver $permissionResolver)
    {
        $this->permissionResolver = $permissionResolver;
    }
}
```
you'll actually inject your decorator.